templates/worker/00-worker/baremetal/files/NetworkManager-resolv-prepender.yaml

```diff
@@ -0,0 +1,27 @@
+filesystem: "root"
+mode: 0755
+path: "/etc/NetworkManager/dispatcher.d/30-resolv-prepender"
+contents:
+  inline: |
+    #!/bin/bash
+    IFACE=$1
+    STATUS=$2
+    case "$STATUS" in
+        up|down|dhcp4-change|dhcp6-change)
+        logger -s "NM resolv-prepender triggered by ${1} ${2}."
+        NAMESERVER_IP=$(/usr/local/bin/non_virtual_ip \
+            "{{.Infra.Status.PlatformStatus.BareMetal.APIServerInternalIP}}" \
+            "{{.Infra.Status.PlatformStatus.BareMetal.NodeDNSIP}}" \
+            "{{.Infra.Status.PlatformStatus.BareMetal.IngressIP}}")
+        set +e
+        if [[ -n "$NAMESERVER_IP" ]]; then
+            logger -s "NM resolv-prepender: Prepending 'nameserver $NAMESERVER_IP' to other nameservers in /var/run/NetworkManager/resolv.conf"
```

Comment:
The log says "to other nameservers in /var/run/..." but this actually updates /etc/resolv.conf

Reply:
Ah good point, thanks - will update!

Reply:
I think it's just worded incorrectly, should say prepending nameserver ... to other nameservers *from* /var/run/...

I'll reword it to clarify we're reading one file and writing the other

Reply:
Ok I pushed an additional commit clarifying the log message